custom_components/audiconnect/dashboard.py

```diff
@@ -390,178 +409,227 @@ def create_instruments():
+        Sensor(
+            attr="tank_level",
+            name="Tank level",
+            icon="mdi:gauge",
+            unit=PERCENTAGE,
+        ),
         Sensor(
             attr="state_of_charge",
             name="State of charge",
             icon="mdi:ev-station",
-            unit="%",
+            unit=PERCENTAGE,
         ),
         Sensor(
             attr="remaining_charging_time",
             name="Remaining charge time",
             icon="mdi:battery-charging",
-            unit=None,
         ),
-        Sensor(attr="plug_state", name="Plug state", icon="mdi:power-plug", unit=None),
+        Sensor(
+            attr="plug_state",
+            name="Plug state",
+            icon="mdi:ev-plug-type1",
+        ),
         Sensor(
             attr="plug_lock_state",
             name="Plug Lock state",
-            icon="mdi:power-plug",
-            unit=None,
+            icon="mdi:ev-plug-type1",
         ),
         Sensor(
             attr="external_power",
             name="External Power",
             icon="mdi:ev-station",
-            unit=None,
         ),
         Sensor(
             attr="plug_led_color",
             name="Plug LED Color",
-            icon="mdi:power-plug",
-            unit=None,
+            icon="mdi:ev-plug-type1",
         ),
         Sensor(
             attr="doors_trunk_status",
             name="Doors/trunk state",
             icon="mdi:car-door",
-            unit=None,
         ),
         Sensor(
             attr="climatisation_state",
             name="Climatisation state",
             icon="mdi:air-conditioner",
-            unit=None,
         ),
         Sensor(
             attr="outdoor_temperature",
             name="Outdoor Temperature",
             icon="mdi:temperature-celsius",
-            unit="°C",
+            unit=UnitOfTemperature.CELSIUS,
+            device_class=SensorDeviceClass.TEMPERATURE,
         ),
         Sensor(
             attr="preheater_duration",
             name="Preheater runtime",
             icon="mdi:clock",
-            unit="Min",
+            unit=UnitOfTime.MINUTES,
         ),
         Sensor(
             attr="preheater_remaining",
             name="Preheater remaining",
             icon="mdi:clock",
-            unit="Min",
+            unit=UnitOfTime.MINUTES,
+        ),
+        BinarySensor(
+            attr="sun_roof",
+            name="Sun roof",
+            device_class=None,
         ),
-        BinarySensor(attr="sun_roof", name="Sun roof", device_class="window"),
         BinarySensor(
             attr="parking_light",
             name="Parking light",
             device_class="safety",
             icon="mdi:lightbulb",
         ),
-        BinarySensor(attr="any_window_open", name="Windows", device_class="window"),
+        BinarySensor(
+            attr="any_window_open",
+            name="Windows",
+            device_class="window",

```

Comment:
This would be BinarySensorDeviceClass.WINDOW imported from 

`from homeassistant.components.binary_sensor import (
    BinarySensorDeviceClass,
    BinarySensorEntity,
    BinarySensorEntityDescription,
)`

Reply:
I stuck to just cleaning up the sensors @kolbi started with. Ran out of time and the entire file needs cleanup. Larger cleanup effort might be best under another PR.

Let me know

Reply:
binary_sensors are done